query/network/network_security_group_remote_access_restricted.sql

```diff
@@ -0,0 +1,41 @@
+with network_sg as (
+  select
+    distinct name sg_name
+  from
+    azure_network_security_group nsg,
+    jsonb_array_elements(security_rules) sg,
+    jsonb_array_elements_text(sg -> 'properties' -> 'destinationPortRanges' || (sg -> 'properties' -> 'destinationPortRange') :: jsonb) dport,
+    jsonb_array_elements_text(sg -> 'properties' -> 'sourceAddressPrefixes' || (sg -> 'properties' -> 'sourceAddressPrefix') :: jsonb) sip
+  where
+    sg -> 'properties' ->> 'access' = 'Allow'
+    and sg -> 'properties' ->> 'direction' = 'Inbound'
+    and sg -> 'properties' ->> 'protocol' = 'TCP'
+    and sip in ('*', '0.0.0.0', '0.0.0.0/0', 'Internet', 'any', '<nw>/0', '/0')
+    and (
+      dport in ('22', '3389', '*')
+      or (
+        dport like '%-%'
+        and split_part(dport, '-', 1) :: integer <= 3389
+        and split_part(dport, '-', 2) :: integer >= 3389
```

Comment:
why we are not handling this part for port `22` ?

Reply:
port 22 added on this part as well.